doc/compiler/optimizer/ControlFlowOptimizationsSummary.md

```diff
@@ -0,0 +1,65 @@
+<!--
+Copyright (c) 2022, 2022 IBM Corp. and others
+
+This program and the accompanying materials are made available under
+the terms of the Eclipse Public License 2.0 which accompanies this
+distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+or the Apache License, Version 2.0 which accompanies this distribution and
+is available at https://www.apache.org/licenses/LICENSE-2.0.
+
+This Source Code may also be made available under the following
+Secondary Licenses when the conditions for such availability set
+forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+General Public License, version 2 with the GNU Classpath
+Exception [1] and GNU General Public License, version 2 with the
+OpenJDK Assembly Exception [2].
+
+[1] https://www.gnu.org/software/classpath/license.html
+[2] http://openjdk.java.net/legal/assembly-exception.html
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+-->
+
+# Control Flow Optimizations
+
+## Block Reordering
+Block Reordering moves blocks around based on making the ‘hot’ path (based on
+profiling and/or static criteria like loop nesting depth, expected code synergy)
+the fall through.
+
+## Block Hoisting
+Block Hoisting duplicates and places the code in a block into its predecessors
+that only have the block as a unique successor, if this would enable better/more
+commoning or other optimizations to occur.
```

Comment:
I would add one extra sentence : `One example of when this optimization may be effective is with respect to local dead store elimination when the hoisted block ends in a return, since might allow the removal of trailing stores to local variables.`